### PR TITLE
Updates for maker_utils API changes

### PR DESCRIPTION
**Pull request by WilliamJamieson**
This PR updates `romancal` so that it is compatible with spacetelescope/roman_datamodels#207. These updates are mostly due to the enforcement of keyword only arguments to the maker_utils. This change was done to unify the API across all the different maker_utils.

**Checklist**
- [x] added entry in `CHANGES.rst` under the corresponding subsection
- [x] updated relevant tests
- [x] updated relevant documentation
- [x] updated relevant milestone(s)
- [x] added relevant label(s)

Regression tests do pass https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/231/

### CODE REVIEW

**Comment by WilliamJamieson** [comment]
I'm leaving this as a draft until the `roman_datamodels` PR is merged, I will then fix the branch it is pointing at and this will be ready to be merged.

**Comment by WilliamJamieson** [comment]
This PR needs to wait until spacetelescope/roman_datamodels#207 is merged so that I can drop the temporary commit.

**Review by ddavis-stsci — APPROVED** [review]

**Pull request merged**
